modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm

```diff
@@ -23,68 +23,233 @@
     "Others find their way into mostly-vampiric communities, turning others into their own kind; though, the virus can only transmit to hosts that are incredibly low on blood, taking advantage of their reduced immune system efficiency and higher rate of blood creation to be able to survive the initial few days within their host.",
     "\"What the fuck does any of this mean?\" - Doctor Micheals, reading their CentCom report about the new 'hires'.")
 
+
 /datum/species/hemophage/prepare_human_for_preview(mob/living/carbon/human/human)
 	human.skin_tone = "albino"
 	human.hair_color = "#1d1d1d"
 	human.hairstyle = "Pompadour (Big)"
 	human.update_mutant_bodyparts(TRUE)
 	human.update_body(TRUE)
+
+
+/obj/item/organ/internal/tongue/hemophage
+	name = "hemophage tongue"
+	color = "#333333"
+	actions_types = list(/datum/action/item_action/organ_action/hemophage)
+	COOLDOWN_DECLARE(drain_cooldown)
+
+
+/datum/action/item_action/organ_action/hemophage
+	name = "Drain Victim"
+	desc = "Leech blood from any carbon victim you are passively grabbing."
+
+/// Maximum an Hemophage will drain, they will drain less if they hit their cap.
+#define HEMOPHAGE_DRAIN_AMOUNT 50
+
+
+/datum/action/item_action/organ_action/hemophage/Trigger(trigger_flags)
+	. = ..()
+	if(iscarbon(owner))
+		var/mob/living/carbon/hemophage = owner
+		var/obj/item/organ/internal/tongue/hemophage/tongue = target
+
+		if(!COOLDOWN_FINISHED(tongue, drain_cooldown))
+			to_chat(hemophage, span_warning("You just drained blood, wait a few seconds!"))
+			return
+
+		if(hemophage.pulling && iscarbon(hemophage.pulling))
+			var/mob/living/carbon/victim = hemophage.pulling
+			if(hemophage.blood_volume >= BLOOD_VOLUME_MAXIMUM)
+				to_chat(hemophage, span_warning("You're already full!"))
+				return
+
+			if(victim.stat == DEAD)
+				to_chat(hemophage, span_warning("You need a living victim!"))
+				return
+
+			if(!victim.blood_volume || (victim.dna && ((NOBLOOD in victim.dna.species.species_traits) || victim.dna.species.exotic_blood)))
+				to_chat(hemophage, span_warning("[victim] doesn't have blood!"))
+				return
+
+
+			var/blood_volume_difference = BLOOD_VOLUME_MAXIMUM - hemophage.blood_volume //How much capacity we have left to absorb blood
+			// We start by checking that the victim is a human and they have a client, so we can give them the
+			// beneficial status effect for drinking higher-quality blood.
+			var/is_target_human_with_client = istype(victim, /mob/living/carbon/human) && victim.client
+
+			if(ismonkey(victim))
+				if(hemophage.blood_volume >= BLOOD_VOLUME_NORMAL)
+					to_chat(hemophage, span_warning("Their inferior blood cannot sate you any further!"))
+					return
+
+				is_target_human_with_client = FALSE // Sorry, not going to get the status effect from monkeys, even if they have a client in them.
```

Comment:
I would argue they should get a status condition-- one that is the opposite of good quality blood, one that makes them have increased blood loss.

Reply:
They're already capped to 560u of blood like the other means of getting blood, increasing the blood drain would literally make it pointless to feed from them, which isn't the goal of this change. The goal is to promote the wanted outcome, rather than heavily punish taking any alternative route.